components/Util/NumericInput.tsx

```diff
@@ -0,0 +1,40 @@
+export default function NumericInput({
+  label,
+  name,
+  max,
+  min,
+  step,
+  defaultValue,
+  value,
+  onChange,
+}: {
+  label: string
+  name?: string
+  max?: number
+  min?: number
+  step?: number
+  defaultValue?: number
+  value?: number
+  onChange?: (evt: any) => void
+}) {
+  if (value && defaultValue) {
+    throw new Error(
+      'Both value and defaultValue were given. Please provide one.'
+    )
+  }
+
+  return (
+    <input
+      type="number"
+      name={name}
+      max={max}
+      min={min}
+      step={step}
+      //For uncontrolled component
+      defaultValue={defaultValue ?? undefined}
+      //For Controlled Component
+      value={value ?? undefined}
+      onChange={onChange ?? undefined}
```

Comment:
Out of all events, `onChange` seems the most logical one to handle first.
The other events can be handled later, as they will become relevant for the `NumericInput` component.

Reply:
I feel we also want to handle `onInput` at least

Reply:
According to [w3schools](https://www.w3schools.com/jsref/event_oninput.asp#:~:text=The%20oninput%20event%20is%20similar,when%20the%20element%20loses%20focus.)

> The oninput event is similar to the [onchange](https://www.w3schools.com/jsref/event_onchange.asp) event.
The difference is that the oninput event occurs immediately after the content has been changed, while onchange occurs when the element loses focus.

Since they are so similar in behavior, tackling both might unnecessarily increase complexity.